openpype/hosts/maya/plugins/publish/extract_thumbnail.py

```diff
@@ -27,22 +29,46 @@ def process(self, instance):
 
         camera = instance.data["review_camera"]
 
-        maya_setting = instance.context.data["project_settings"]["maya"]
-        plugin_setting = maya_setting["publish"]["ExtractPlayblast"]
-        capture_preset = plugin_setting["capture_preset"]
+        host_name = instance.context.data["hostName"]
+        family = instance.data["family"]
+        task_data = instance.data["anatomyData"].get("task", {})
+        task_name = task_data.get("name")
+        task_type = task_data.get("type")
+        subset = instance.data["subset"]
```

Comment:
These appear to be used once, and only in the `filtering_criteria`. I'd either do:

1. Move the `get_profile` logic into a method on the class so there's less clutter in the code and easier to see where the variables are used.
2. OR just put these directly into the filtering_criteria dict so it's e.g. `"hosts": context.data["hostName"]`

Reply:
Cool, will condense.